pkg/messagebus/producer.go

```diff
@@ -0,0 +1,27 @@
+package messagebus
+
+import (
+	"context"
+	"time"
+)
+
+type (
+	Producer interface {
+		Reader
+		Subscriber
+		Poller
+		Storer
+	}
+
+	Subscriber interface {
```

Comment:
Why not a Notifier? Or should fn. be named Subscribe? :)

Reply:
refactored